syntax = "proto3";

//protoc --python_out=. message_type.proto
message COMM_MESSAGE {
	enum TYPE {
		CONFIRM = 0;
		LOGIN = 1;
		LIST = 2;
		LIST_PART2 = 3;
		CLIENT_TO_CLIENT = 4;
		MESSAGE = 5;
		LOGOUT = 6;
		ERROR = 7;
	}
	TYPE type = 1;
	string N1 = 2;
	string N1_hash = 3;
	string gb_mod_p = 4;
	bytes iv = 5;
	bytes tag = 6;
	string message = 7;
	bytes cipher_text = 8;
	bytes ticket = 9;
	bytes ticket_tag = 10;
}
